rust/sdk-processor/src/steps/parquet_default_processor/parquet_default_extractor.rs

```diff
@@ -0,0 +1,146 @@
+        debug!("Processed data sizes:");
+        debug!(" - MoveResources: {}", move_resources.len());
+        debug!(" - WriteSetChanges: {}", write_set_changes.len());
+        debug!(" - ParquetTransactions: {}", parquet_transactions.len());
+        debug!(" - TableItems: {}", table_items.len());
+        debug!(" - MoveModules: {}", move_modules.len());
+
+        let mut map: HashMap<ParquetTypeEnum, ParquetTypeStructs> = HashMap::new();
+
+        // Array of tuples for each data type and its corresponding enum variant
+        let data_types = [
+            (
+                ParquetTypeEnum::MoveResource,
+                ParquetTypeStructs::MoveResource(move_resources),
+            ),
+            (
+                ParquetTypeEnum::WriteSetChange,
+                ParquetTypeStructs::WriteSetChange(write_set_changes),
+            ),
+            (
+                ParquetTypeEnum::Transaction,
+                ParquetTypeStructs::Transaction(parquet_transactions),
+            ),
+            (
+                ParquetTypeEnum::TableItem,
+                ParquetTypeStructs::TableItem(table_items),
+            ),
+            (
+                ParquetTypeEnum::MoveModule,
+                ParquetTypeStructs::MoveModule(move_modules),
+            ),
+        ];
+
+        // Populate the map based on opt-in tables
+        for (enum_type, data) in data_types {
+            add_to_map_if_opted_in_for_backfill(&self.opt_in_tables, &mut map, enum_type, data);
+        }
+
+        debug!(
+            "Map populated with data for the following tables: {:?}",
+            map.keys().collect::<Vec<_>>()
+        );
+
+        Ok(Some(TransactionContext {
+            data: map,
+            metadata: transactions.metadata,
+        }))
+    }
+}
+
+pub fn process_transactions(
+    transactions: Vec<Transaction>,
+) -> (
+    Vec<MoveResource>,
+    Vec<WriteSetChangeModel>,
+    Vec<TransactionModel>,
+    Vec<TableItem>,
+    Vec<MoveModule>,
+) {
+    // this will be removed in the future.
+    let mut transaction_version_to_struct_count = AHashMap::new();
+    let (txns, _, write_set_changes, wsc_details) = TransactionModel::from_transactions(
+        &transactions,
+        &mut transaction_version_to_struct_count,
+    );
+
+    let mut move_modules = vec![];
+    let mut move_resources = vec![];
+    let mut table_items = vec![];
+
+    for detail in wsc_details {
+        match detail {
+            WriteSetChangeDetail::Module(module) => {
+                move_modules.push(module);
+            },
+            WriteSetChangeDetail::Resource(resource) => {
+                move_resources.push(resource);
+            },
+            WriteSetChangeDetail::Table(item, _, _) => {
+                table_items.push(item);
+            },
+        }
+    }
+
+    (
+        move_resources,
+        write_set_changes,
+        txns,
+        table_items,
+        move_modules,
+    )
+}
+
```

Comment:
Is it possible to re-use the `process_transactions` from the Parquet default processor?

Reply:
Or better yet, resuse this one from the Postgres default processor: https://github.com/aptos-labs/aptos-indexer-processors/blob/7b2873f722cc3a1d5d9856378483cbb3aa4e0222/rust/processor/src/processors/default_processor.rs#L295

Reply:
yeah we can reuse the the process_transactions from the parquet default processor!

just sharing the context why initially I decided  not to re-use the `process_transactions` from the parquet processor because there is that  we wouldn't need  the `struct count` map. having new code in this file allows us to simply remove the code later after migrated all. **but** it already re-uses some of the functions from the existing parquet processor lol. 

> Or better yet, resuse this one from the Postgres default processor:
I don't think we can reuse the current version of `process_transactions` from the postgres default processor since they are kind of diverged due to Parquet migration.

Keeping them as separate functions could simplify the code by avoiding conditionals for deprecated tables, making each version cleaner and more maintainable. Alternatively, if there’s significant overlap, we could refactor shared logic into helper functions. What do you think—is it better to keep them separate or refactor shared portions?

Reply:
Just to understand the context: what part of the Postgres and Parquet diverged during the migration?

Reply:
unresolving, also curious about the above ^ @yuunlimm